src/main/java/com/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -1,17 +1,42 @@
 package com.bookstore.service.impl;
 
+import com.bookstore.dto.BookDto;
+import com.bookstore.dto.CreateBookRequestDto;
+import com.bookstore.exception.EntityNotFoundException;
+import com.bookstore.mapper.BookMapper;
 import com.bookstore.model.Book;
+import com.bookstore.repository.BookRepository;
 import com.bookstore.service.BookService;
 import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
 
+@RequiredArgsConstructor
+@Service
 public class BookServiceImpl implements BookService {
+
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto createBookRequestDto) {
+        Book book = bookMapper.toModel(createBookRequestDto);
+        Book savedBook = bookRepository.save(book);
+        return bookMapper.toDto(savedBook);
+    }
+
     @Override
-    public Book save(Book book) {
-        return null;
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return null;
+    public BookDto getByBookId(Long id) {
+        Book book = bookRepository.getByBookId(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book doesn't exist. ID: " + id));
+
+        return bookMapper.toDto(book);
```

Comment:
Please refactor this method to not store the `book` variable and return results immediately - use `Optional.map()` method.